src/client-functions/selectors/add-api.js

```diff
@@ -268,6 +269,86 @@ function createDerivativeSelectorWithFilter (getSelector, SelectorBuilder, selec
     return builder.getFunction();
 }
 
+var ensureRegExpValue = str => typeof str === 'string' ? new RegExp(escapeRe(str)) : str;
+
+function sandboxed (fn) {
+    var code = `(${fn.toString()})`;
+
+    code = processScript(code, false);
+
+    /* eslint-disable no-eval */
+    return eval(code, null);
+    /* eslint-enable no-eval */
+}
+
+var getAttrs     = sandboxed(element => element.attributes);
+var getInnerText = sandboxed(element => element.innerText);
+
+/* eslint-disable no-undef */
+function hasText (node) {
```

Comment:
If those method are used in ClientFunctions they are better be declared as ClientFunctions, otherwise code recompilation and sandboxing will happen on every use.